gpio/__init__.py

```diff
@@ -20,75 +20,98 @@
 #########################################################################
 
 import logging
-from lib.model.smartplugin import SmartPlugin
-import RPi.GPIO as GPIO
+from lib.model.smartplugin import *
 import threading
+import datetime
+
+try:
+    import RPi.GPIO as GPIO
+    REQUIRED_PACKAGE_IMPORTED = True
+except:
+    REQUIRED_PACKAGE_IMPORTED = False
 
 class Raspi_GPIO(SmartPlugin):
-    PLUGIN_VERSION = "1.0.1"
+    PLUGIN_VERSION = "1.4.1"
     ALLOW_MULTIINSTANCE = False
 
-    def __init__(self, sh, mode="board"):
+    def __init__(self, sh):
         self.logger = logging.getLogger(__name__)
-        self._sh = sh
-        self._items = []
-        self._itemsdict = {}
-        self._mode = mode.upper()
-        GPIO.setwarnings(False)
-        if self._mode == "BCM":
-            GPIO.setmode(GPIO.BCM)
-        else:
-            GPIO.setmode(GPIO.BOARD)    
-        self.logger.debug("GPIO: Mode set to {0}".format(self._mode))
-        self.alive = False
-        self._lock = threading.Lock() 
-        
+        self.init_webinterface()
+        self._name = self.get_fullname()
+        if not REQUIRED_PACKAGE_IMPORTED:
+            self.logger.error("{}: Unable to import Python package 'GPIO'".format(self._name))
+            self._init_complete = False
+            return
```

Comment:
I don't know if this error handling will help people or not. Sure, this prevents a fatal error, but also does not handle such an incomplete plugin setup in other places of this plugin.

Someone is still able to call the `send()` method directly ending with a fatal error. Further an incomplete setup will not avoid that the `parse_item ()` will be called for all the item - which is senseless dude like since the GPIO can not be used.

Reply:
It's taken from the sample plugin...

Reply:
Isn't it the same as described here? https://www.smarthomeng.de/fehlerbehandlung-fuer-plugins-bei-nicht-installierten-python-packages